.circleci/config.yml

```diff
@@ -17,19 +17,23 @@ jobs:
     steps:
       - checkout
 
+      - run:
```

Comment:
Added some proper caching based on dependencies changing to help speed up CI. I was running into problems with timeouts so thought it would be food to fix while I'm here.